### PR TITLE
fix forward and backward for norm with negative infinity norm 

### PR DESCRIPTION
**Pull request by Separius**
I found a bug in norm() and fixed it (and added tests to make sure it's fixed)
here is how to reproduce it:
```python
import torch
x = torch.FloatTensor([[10, 12, 13], [4, 0, 12]])
print(torch.norm(x, -40, dim=0, keepdim=True)) #output is tensor([[ 4.0000,  0.0000, 11.9853]])
print(torch.norm(x, float('-inf'), dim=0, keepdim=True)) #output is tensor([[1., 1., 1.]]) which is wrong!
from numpy.linalg import norm as np_norm
x = x.numpy()
print(np_norm(x, ord=-40, axis=0)) #output is array([[4., 0., 11.985261]])
print(np_norm(x, ord=float('-inf'), axis=0)) #output is array([[4., 0., 12.0]])
```
it's related to [#6817](https://github.com/pytorch/pytorch/issues/6817) and [#6969](https://github.com/pytorch/pytorch/pull/6969)

### CODE REVIEW

**Review by ssnl — APPROVED** [review]
Nice! Code looks very reasonable. Thank you! Our CI seems to be having some problem. Could you rebase to master and force push? Sorry about it!

**Comment by Separius** [comment]
@SsnL I checked CI's log and there were two problems, one is with data_loader which is probably unrelated to my code, but the other one was a jit test that my code failed, turns out there is a function in test_jit.py named [get_constants()](https://github.com/pytorch/pytorch/blob/04d4ec285c8452bea0dfc38d98d0edebe6fe339e/test/test_jit.py#L8119) and I think it has a bug(an obvious one), but maybe(0.001%) it was intentional, can you check that too?; I will rebase after that
one more thing is it better to squash all my commits into one or two is better in this case? (jit test; norm)

**Review by ssnl — APPROVED** [review]
Oh Thanks! I must admit that I didn't look at the log and assumed that it's the CI base commit problem (rebase usually solves that). My bad! We recently switched from jenkins to CircleCI so there are still some rough edges around CI. Thanks for fixing the JIT issue!

**Comment by Separius** [comment]
rebased and squashed all commits into one, ready to merge! :) @SsnL 
BTW: thanks for your fast replies

File: torch/csrc/jit/register_prim_ops.cpp [hunk 1]
```diff
@@ -171,13 +171,15 @@ RegisterOperators reg({
...
           return [](Stack& stack) {
             auto s = pop(stack).toString();
-            if (s->string() != "inf") {
+            if (s->string() == "inf")
+              push(stack, std::numeric_limits<double>::infinity());
+            else if (s->string() == "-inf")
+              push(stack, -std::numeric_limits<double>::infinity());
+            else
               AT_ERROR(
                   "Only 'inf' can be cast to a float, but got '",
```
Comment (zou3519) [inline torch/csrc/jit/register_prim_ops.cpp]: nit: change this to "only 'inf' or '-inf' can be casted..."

**Review by zou3519 — COMMENTED** [review]
jit changes look good, but please add a test for -inf to https://github.com/pytorch/pytorch/blob/7edfe11ba49c8539ee904f06320050916857c8fe/test/test_jit.py#L2811-L2816

**Comment by Separius** [comment]
@zou3519 fixed the error message and added a test for -inf

**Review by soumith — APPROVED** [review]

**Pull request closed without merging**